docs/content/2.deploy/providers/firebase.md

```diff
@@ -82,11 +104,92 @@ NITRO_PRESET=firebase yarn build
 firebase emulators:start
 ```
 
-## Deploy to Firebase Hosting via CLI
+## Build and Deploy
 
-Deploying to Firebase Hosting is a simple matter of just running the `firebase deploy` command.
+Deploy to Firebase Hosting by running a nitro build and then running the `firebase deploy` command.
 
 ```bash
 NITRO_PRESET=firebase yarn build
+```
+
+```bash
+npx firebase-tools deploy
+```
+
+If you chose the alternative setup using the Firebase CLI, you can also run:
+
+```bash
 firebase deploy
 ```
+
+## Firebase Functions v2
+
+**Preset:** `firebase-v2` ([switch to this preset](/deploy/#changing-the-deployment-preset))
+
+This preset will deploy to [firebase functions V2](https://firebase.google.com/docs/functions/version-comparison) - the 2nd generation of cloud functions.
+
+It works the same as the [`firebase`](#firebase) preset above, but uses the V2 functions runtime.
+
+### Setup
+
+Follow the instructions for the `firebase` preset above for setup and deploy.
+
+**Caution**: New HTTP and HTTP callable functions deployed with any Firebase CLI lower than version 7.7.0 are private by default and throw HTTP 403 errors when invoked. Either explicitly [make these functions public](https://cloud.google.com/functions/docs/securing/managing-access-iam#allowing_unauthenticated_http_function_invocation) or [update your Firebase CLI](https://firebase.google.com/docs/cli#setup_update_cli) before you deploy. If you deploy the function and it is not publicly accessible, then you will need to visit the [Functions Console](https://console.cloud.google.com/functions/list) on Google cloud platform, click your function name, select the 'Permissions' tab and follow the instructions on how to make the function public.
```

Comment:
It might be worth making this warning a bit more noticeable. Example [here](https://nitro.unjs.io/deploy/providers/bun). 

```suggestion
::alert{type="warning"}
New HTTP and HTTP callable functions deployed with any Firebase CLI lower than version 7.7.0 are private by default and throw HTTP 403 errors when invoked. Either explicitly [make these functions public](https://cloud.google.com/functions/docs/securing/managing-access-iam#allowing_unauthenticated_http_function_invocation) or [update your Firebase CLI](https://firebase.google.com/docs/cli#setup_update_cli) before you deploy. If you deploy the function and it is not publicly accessible, then you will need to visit the [Functions Console](https://console.cloud.google.com/functions/list) on Google cloud platform, click your function name, select the 'Permissions' tab and follow the instructions on how to make the function public.
::
```